# Airbnb Amenities Analysis
⌨ Jeran Burget

This analysis explores different amenities offered in Airbnb listings in Salem, Oregon. The objective of this analysis is to determine which amenities in Salem are the most common, and which ones are the least commonly observed.

## About the data
This data set `airbnb.csv` comes from the Airbnb website and describes about 300 Airbnb listings in the Salem, Oregon area. The data describes many aspects of the listings including information about the hosts, listing description, and amenities offered at the facilities.

## Analysis
The goal of this analysis is to determine which amenities are the most common and which are the least common in the Salem, Oregon area. However, because the amenities are stored as lists in a single column of the data set, this information is difficult to analyze without the use of for loops.

In [3]:
import pandas as pd
df = pd.read_csv('airbnb.csv')

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 302 entries, 0 to 301
Data columns (total 19 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   name                  302 non-null    object 
 1   description           302 non-null    object 
 2   host_name             302 non-null    object 
 3   host_since            302 non-null    object 
 4   host_location         220 non-null    object 
 5   host_about            176 non-null    object 
 6   host_response_time    273 non-null    object 
 7   host_response_rate    273 non-null    object 
 8   host_acceptance_rate  278 non-null    object 
 9   host_verifications    302 non-null    object 
 10  property_type         302 non-null    object 
 11  room_type             302 non-null    object 
 12  accommodates          302 non-null    int64  
 13  bathrooms             0 non-null      float64
 14  bathrooms_text        302 non-null    object 
 15  bedrooms              1

In [13]:
counter = {}

for i, row in df.iterrows():
    for amenity in eval(row['amenities']):
        if amenity in counter:
             counter[amenity] += 1
        else:
            counter[amenity] = 1
counter

{'Host greets you': 27,
 'Baby safety gates': 11,
 'Essentials': 269,
 'Carbon monoxide alarm': 275,
 'Pack ’n play/Travel crib': 44,
 'Single level home': 67,
 'Long term stays allowed': 172,
 'Refrigerator': 230,
 'Blender': 80,
 'Dryer': 75,
 'Coffee maker: Keurig coffee machine, pour-over coffee': 3,
 'Iron': 221,
 'Rice maker': 44,
 'Hair dryer': 219,
 'Drying rack for clothing': 27,
 'Books and reading material': 71,
 'Fire extinguisher': 256,
 'Clothing storage: closet and dresser': 54,
 'Luggage dropoff allowed': 97,
 'Toaster': 162,
 'Fireplace guards': 11,
 'Microwave': 247,
 'Freezer': 169,
 'Portable fans': 88,
 'Cooking basics': 229,
 'Indoor fireplace: gas': 33,
 'Extra pillows and blankets': 196,
 'Barbecue utensils': 56,
 'BBQ grill': 78,
 'Kayak': 1,
 'Body soap': 120,
 'Smoke alarm': 301,
 'Private backyard – Fully fenced': 67,
 'Children’s dinnerware': 32,
 'Stove': 78,
 'TV': 120,
 'First aid kit': 236,
 'Free driveway parking on premises': 8,
 'AC - split type duct

To make this analysis even easier to read, let's loop through each amenity in the counter and only print out the ones that have more than 200 occurences.

In [14]:
for amenity in counter:
    if counter[amenity] > 200:
        print(amenity, counter[amenity]) 

Essentials 269
Carbon monoxide alarm 275
Refrigerator 230
Iron 221
Hair dryer 219
Fire extinguisher 256
Microwave 247
Cooking basics 229
Smoke alarm 301
First aid kit 236
Dishes and silverware 260
Wifi 252
Hangers 259
Hot water 245
Kitchen 272
Bed linens 234
Free parking on premises 238
Self check-in 206


## Conclusion
Out of the 302 Airbnb listings, here are the most common amenities:

`Essentials,
Carbon monoxide alarm,
Refrigerator,
Iron,
Hair dryer,
Fire extinguisher,
Microwave,
Cooking basics, 
Smoke alarm, 
First aid kit, 
Dishes and silverware, 
Wifi, 
Hangers, 
Hot water, 
Kitchen, 
Bed linens, 
Free parking on premises, 
Self check-in`

While not every listing will have all of these items, you'd be hard pressed to find listings without a lot of these.